hal/halx86/acpi/halacpi.c

```diff
@@ -35,15 +36,339 @@ PACPI_BIOS_MULTI_NODE HalpAcpiMultiNode;
+PPM_DISPATCH_TABLE PmAcpiDispatchTable;
+
+/* DISPATCH TABLE FUNCTIONS ***************************************************/
+
+ACPI_PM_DISPATCH_TABLE HalAcpiDispatchTable =
+{
+    'HAL ',
+    2,
+    HaliAcpiTimerInit,
+    NULL,
+    HaliAcpiMachineStateInit,
+    HaliAcpiQueryFlags,
+    HalpAcpiPicStateIntact,
+    HalRestorePicState,
+    HaliPciInterfaceReadConfig,
+    HaliPciInterfaceWriteConfig,
+    HaliSetVectorState,
+    HalSystemVector,
+    HaliSetMaxLegacyPciBusNumber,
+    HaliIsVectorValid
+};
+
+VOID
+NTAPI
+HaliAcpiMachineStateInit(_In_ ULONG Unknown1,
+                         _In_ PVOID State,
+                         _Out_ ULONG * OutInterruptModel)
+{
+    //UNIMPLEMENTED;
+    DPRINT1("HaliAcpiMachineStateInit: .. \n");
+    ASSERT(FALSE);
+}
+
+NTSTATUS
+NTAPI
+HaliAcpiQueryFlags(_In_ ULONG Unknown1,
+                   _In_ ULONG Unknown2,
+                   _In_ ULONG Unknown3)
+{
+    //UNIMPLEMENTED;
+    DPRINT1("HaliAcpiQueryFlags: .. \n");
+    ASSERT(FALSE);
+    return STATUS_NOT_IMPLEMENTED;
+}
+
+NTSTATUS
+NTAPI
+HalpAcpiPicStateIntact(_In_ ULONG Unknown1,
+                       _In_ ULONG Unknown2,
+                       _In_ ULONG Unknown3)
+{
+    //UNIMPLEMENTED;
+    DPRINT1("HalpAcpiPicStateIntact: .. \n");
+    ASSERT(FALSE);
+    return STATUS_NOT_IMPLEMENTED;
+}
+
+NTSTATUS
+NTAPI
+HalRestorePicState(_In_ ULONG Unknown1,
+                   _In_ ULONG Unknown2,
+                   _In_ ULONG Unknown3)
+{
+    //UNIMPLEMENTED;
+    DPRINT1("HalRestorePicState: .. \n");
+    ASSERT(FALSE);
+    return STATUS_NOT_IMPLEMENTED;
+}
+
+UCHAR
+NTAPI
+HaliSetVectorState(_In_ ULONG GlobalIrq,
+                   _In_ UCHAR State)
+{
+    //UNIMPLEMENTED;
+    DPRINT1("HaliSetVectorState: GlobalIrq %X, State %X\n", GlobalIrq, State);
+    ASSERT(FALSE);
+    return 0;
+}
+
+NTSTATUS
+NTAPI
+HalSystemVector(_In_ ULONG Unknown1,
+                _In_ ULONG Unknown2,
+                _In_ ULONG Unknown3)
+{
+    //UNIMPLEMENTED;
+    DPRINT1("HalSystemVector: .. \n");
+    ASSERT(FALSE);
+    return STATUS_NOT_IMPLEMENTED;
+}
+
+VOID
+NTAPI
+HaliSetMaxLegacyPciBusNumber(_In_ ULONG MaxLegacyPciBusNumber)
+{
+    //UNIMPLEMENTED;
+    DPRINT1("HaliSetMaxLegacyPciBusNumber: MaxLegacyPciBusNumber %X\n", MaxLegacyPciBusNumber);
+    ASSERT(FALSE);
+}
+
+BOOLEAN
+NTAPI
+HaliIsVectorValid(_In_ ULONG DeviceIrq)
+{
+    //UNIMPLEMENTED;
+    DPRINT1("HaliIsVectorValid: DeviceIrq %X\n", DeviceIrq);
+    ASSERT(FALSE);
+    return FALSE;
+}
+
+/* PM TIMER FUNCTIONS *********************************************************/
+
+VOID
+NTAPI
+HaliPmTimerQueryPerfCount(_Out_ LARGE_INTEGER * OutPerfCount,
+                          _Out_ LARGE_INTEGER * OutPerfFrequency)
+{
+    OutPerfCount->QuadPart = TimerInfo.PerformanceCounter.QuadPart + QueryTimer();
+
+    if (OutPerfFrequency) {
+        OutPerfFrequency->QuadPart = PMTimerFreq;
+    }
+}
+
+ULONGLONG
+__cdecl
+HalpQueryPerformanceCounter(VOID)
```

Comment:
I'd place ACPI PM timer stuff into a separate file. This one is getting big

APIC HAL uses ACPI PM timer instead of TSC counter like we do? Or this is just for ACPI PIC HAL?

I'd say, timer stuff is a big topic by itself. We should implement a common way of switching between clock sources (PM timer, TSC, HPET, PIT). Deserves a separate PR...

Reply:
Now it in 'acpitimer.c'